APIComparer.Contracts/CompareNugetPackage.cs

```diff
@@ -1,20 +1,20 @@
 ﻿// ReSharper disable MemberCanBePrivate.Global
 namespace APIComparer.Contracts
 {
-    using System;
     using NServiceBus;
+    using NuGet;
 
     public class CompareNugetPackage : ICommand
     {
-        public CompareNugetPackage(string packageId, Version leftVersion, Version rightVersion)
+        public CompareNugetPackage(string packageId, SemanticVersion leftVersion, SemanticVersion rightVersion)
         {
             PackageId = packageId;
             RightVersion = rightVersion;
             LeftVersion = leftVersion;
         }
 
-        public string PackageId { get; private set; }
-        public Version LeftVersion { get; private set; }
-        public Version RightVersion { get; private set; }
+        public string PackageId { get; set; }
+        public SemanticVersion LeftVersion { get; set; }
```

Comment:
In the first call we agree on not using SemanticVersion of nuget. If this doesn't have restrictions when we analyze non semantic version compliant packages I'm OK with that change but I agree with @andreasohlund that we shouldn't tie the message contract to the nuget type.